### PR TITLE
[iOS] Map view can now be optionally configured to make pitch adjustments without affecting altitude

### PR DESCRIPTION
**Pull request by d-prukop**
Moved from #12518 

Map view can now be optionally configured to make pitch adjustments without affecting altitude. 

This addresses part of #9808

Pitch changes using MGLAltitudeForZoomLevel and MGLZoomLevelForAltitude from the MGLGeometry library were animating in a strange fashion when using the new proposed camera animation system (located here: https://github.com/mapbox/mapbox-ios-map-camera).

I added a flag to turn off this association between pitch and altitude.

This property is not exposed in the header.

Before | After
before | after

### CODE REVIEW

**Comment by d-prukop** [comment]
@julianrex yes... kind of. I think it leaves us in the situation mentioned in the last sentence of that post. It's not ideal but it should be acceptable until we can rework the camera.

File: platform/ios/src/MGLMapView.mm [hunk 1]
```diff
@@ -250,6 +250,8 @@ @interface MGLMapView () <UIGestureRecognizerDelegate,
 @property (nonatomic) CFTimeInterval frameTime;
 @property (nonatomic) CFTimeInterval averageFrameTime;
 
+@property (nonatomic, getter=isCameraAltitudeAffectedByPitch) BOOL cameraAltitudeAffectedByPitch;
```
Comment (1ec5) [inline platform/ios/src/MGLMapView.mm]: #12966 proposes an alternative to this property that avoids implicitly reinterpreting existing properties.

Comment (julianrex) [reply platform/ios/src/MGLMapView.mm]: @d-prukop in reference to the above https://github.com/mapbox/mapbox-gl-native/pull/12966#issuecomment-425213127, assuming that lands first will this PR require a change?

File: platform/ios/src/MGLMapView.mm [hunk 2]
```diff
@@ -2388,7 +2392,8 @@ - (void)resetPosition
     auto camera = _mbglMap->getStyle().getDefaultCamera();
     CGFloat pitch = *camera.pitch;
     CLLocationDirection heading = mbgl::util::wrap(*camera.angle, 0., 360.);
-    CLLocationDistance distance = MGLAltitudeForZoomLevel(*camera.zoom, pitch, 0, self.frame.size);
+    CGFloat pitchForAltitude = self.isCameraAltitudeAffectedByPitch ? pitch : 0.0;
+    CLLocationDistance distance = MGLAltitudeForZoomLevel(*camera.zoom, pitchForAltitude, 0, self.frame.size);
     self.camera = [MGLMapCamera cameraLookingAtCenterCoordinate:MGLLocationCoordinate2DFromLatLng(*camera.center)
```
Comment (julianrex) [inline platform/ios/src/MGLMapView.mm]: Regardless of whether #12966 lands, we should probably change this to use 
`-[MGLMapCamera cameraLookingAtCenterCoordinate:altitude:pitch:heading:]` and rename the `distance` variable as `altitude`. Is my understanding correct?

File: platform/ios/src/MGLMapView.mm [hunk 3]
```diff
@@ -3464,7 +3470,8 @@ - (MGLMapCamera *)cameraForCameraOptions:(const mbgl::CameraOptions &)cameraOpti
     double zoomLevel = cameraOptions.zoom ? *cameraOptions.zoom : self.zoomLevel;
     CLLocationDirection direction = cameraOptions.angle ? mbgl::util::wrap(-MGLDegreesFromRadians(*cameraOptions.angle), 0., 360.) : self.direction;
     CGFloat pitch = cameraOptions.pitch ? MGLDegreesFromRadians(*cameraOptions.pitch) : _mbglMap->getPitch();
-    CLLocationDistance altitude = MGLAltitudeForZoomLevel(zoomLevel, pitch, centerCoordinate.latitude, self.frame.size);
+    CGFloat pitchForAltitude = self.isCameraAltitudeAffectedByPitch ? pitch : 0.0;
+    CLLocationDistance altitude = MGLAltitudeForZoomLevel(zoomLevel, pitchForAltitude, centerCoordinate.latitude, self.frame.size);
     return [MGLMapCamera cameraLookingAtCenterCoordinate:centerCoordinate fromDistance:altitude pitch:pitch heading:direction];
```
Comment (julianrex) [inline platform/ios/src/MGLMapView.mm]: Same here.

**Comment by 1ec5** [comment]
Closing in favor of #12966 and #12986 per discussion with @d-prukop.

**Pull request closed without merging**